### PR TITLE
feat: maybe_parallel_map util function

### PR DESCRIPTION
**Pull request by pakhomov-dfinity**
In few places we have `Option<&mut Pool>` and then duplicate code for `parallel_map` and sequential for.
This PR introduces a helper function to deduplicate the code.

### CODE REVIEW

**Review by schneiderstefan — APPROVED** [review]
Good idea. We might also want check if we should replace some `Option<Pool>` with just `Pool` at some places. I suspects some of these options are historical leftovers that we don't actually use.

File: rs/utils/src/thread.rs [hunk 1]
```diff
@@ -27,6 +27,24 @@ where
         .collect()
 }
 
+/// parallel_map(...) if thread_pool is Some(); map if None.
```
Comment (stiegerc) [inline rs/utils/src/thread.rs:30]:
```suggestion
/// `parallel_map(_)` if thread_pool is `Some(_)`; map if `None`.
```
pedantic nit.

**Review by stiegerc — APPROVED** [review]
Seems like a nice improvement.

File: rs/utils/src/thread.rs (hunk 1 above)
Comment (alin-at-dfinity) [reply rs/utils/src/thread.rs:30]: Actually:

```suggestion
/// `parallel_map(_)` if thread_pool is `Some`; `map(_)` if `None`.
```

**Review by alin-at-dfinity — APPROVED** [review]

**Pull request merged**